packages/peregrine/lib/talons/Header/useAccountTrigger.js

```diff
@@ -47,24 +56,82 @@ export const useAccountTrigger = props => {
         // would be to call apolloClient.resetStore() but that would require
         // a large refactor.
         history.go(0);
-    }, [apolloClient, history, revokeToken, setAccountMenuIsOpen, signOut]);
+    }, [
+        apolloClient,
+        history,
+        revokeToken,
+        setAccountMenuIsOpen,
+        signOut,
+        VIEWS.SIGNIN
+    ]);
 
     const handleTriggerClick = useCallback(() => {
         // Toggle the Account Menu.
         setAccountMenuIsOpen(isOpen => !isOpen);
-    }, [setAccountMenuIsOpen]);
+        /**
+         * When the dropdown closes, the view has to
+         * reset to SIGNIN view if the user is not
+         * signed in.
+         */
+        if (!isUserSignedIn) {
+            setView(VIEWS.SIGNIN);
+        }
+    }, [setAccountMenuIsOpen, isUserSignedIn, VIEWS]);
+
+    const handleForgotPassword = useCallback(() => {
+        setView(VIEWS.FORGOT_PASSWORD);
+    }, [VIEWS.FORGOT_PASSWORD]);
+
+    const handleCreateAccount = useCallback(() => {
+        setView(VIEWS.CREATE_ACCOUNT);
+    }, [VIEWS.CREATE_ACCOUNT]);
+
+    const handleFormErrors = useCallback(
+        errors => {
+            setFormErrors([...formErrors, ...errors]);
```

Comment:
Ah here is my confusion - why does the trigger care about form errors for some child component?

Reply:
Because of the placement of the errors. All errors need to be shown in a single place on the UI. If we don't do this, SignIn errors will show up in a different place and the errors from other views in Signin / Trigger errors in another place. We do not want to show / render multiple `FormError` components.

To avoid this, during the slack conversation couple of days ago, we decided to try the state lift design.

Reply:
I must have missed the conversation. 

For what it's worth, #2588 will update the sign in component with a new error handling styling. If you're going to re-use the component (I think that's a great idea) then you actually don't need to do any error handling in this PR at all. I think we can back out the error handling changes from this PR and keep the scope smaller.